### ISSUE TITLE
For multi-calss seg, num_class=3 for example

### ISSUE THREAD

**Issue by WangPing521** [WuJunde/MedSegDiff#33]
Hi Junde Wu,

I have some questions for you.

The hyper-parameter in_ch=2 is fixed no matter of binary or multi-class task, where the two dimension includes the image and the mask.
For multi-calss task, what we are supposed to change is only the calibration output, i.e. sigmoid to softmax, then we can get a [1 3 H W] calibration and a [1 2 H W] model_output. Is that correct?

If we change the in_ch = 3 + 1(one-hot with the image condition), we can have the [1 3 H W] calibration, however, i do not know what is the model_output? is it something like [1 3 2 H W]? or it is also the [1 2 H W], if so, using mask rather then one-hot as the input of diffusion model seems to be meaningful?

I grouped a 5-class task into binary case to check the results. Here are one visualization, is it correct? From top to bottom, img, recovery from diffusion model, calibration, linear combination of the recovery and calibration.
![image](https://user-images.githubusercontent.com/47101292/218598638-081e3148-3519-42ac-9ddc-f486474f7050.png)

Thanks!
Ping

**Comment by saisusmitha** [comment]
@WangPing521 Hi. Do you know how and where to add code to find metrics[ hd95,jaccard,dice]? Also is there any validation code? how to find these scores?

**Comment by saisusmitha** [comment]
@WangPing521 Hi. Do you know how and where to add code to find metrics[ hd95,jaccard,dice]? Also is there any validation code? how to find these scores?

**Comment by cannonli7** [comment]
你好，请问unet的初始化参数out_channels设置为2是不是针对二值分割任务？这样设置似乎有一个问题，
def _predict_xstart_from_eps(self, x_t, t, eps):
        assert x_t.shape == eps.shape

在以上函数中eps是unet输出，如果out_channel设置为2，那么eps的shape与x_t的通道数不同？请问out_channels是否需要修改
